src/yunohost/tools.py

```diff
@@ -163,6 +163,10 @@ def tools_maindomain(auth, new_domain=None):
         logger.warning("%s" % e, exc_info=1)
         raise MoulinetteError(errno.EPERM, m18n.n('maindomain_change_failed'))
 
+    # Clear nsswitch cache for hosts to make sure hostname is resolved ...
+    service_regen_conf(['nsswitch'], force=True)
```

Comment:
Moved it back to the postinstall, it wasnt clear to me why I put in the maindomain function 😕